q2_demux/_demux.py

```diff
@@ -48,6 +52,47 @@ def __next__(self):
         return next(self.barcode_generator), next(self.sequence_generator)
 
 
+def summary(output_dir: str, data: SingleLanePerSampleSingleEndFastqDirFmt) \
+        -> None:
+    per_sample_fastqs = list(data.sequences.iter_views(FastqGzFormat))
+    per_sample_fastq_counts = {}
+    for per_sample_fastq in per_sample_fastqs:
+        seqs = skbio.io.read(per_sample_fastq[1].open(),
+                             format='fastq',
+                             phred_offset=33, compression='gzip',
+                             constructor=skbio.DNA)
+        count = 0
+        for seq in seqs:
+            count += 1
+        per_sample_fastq_counts[per_sample_fastq[0]] = count
+
+    result = pd.Series(per_sample_fastq_counts)
+    result.sort_values(inplace=True, ascending=False)
+    result.to_csv(os.path.join(output_dir,
```

Comment:
What does the CSV output file look like? There's some attributes we can set on the Series to make the CSV output more informative (i.e. include a header). Here's what I think we need:

```python
result.name = 'Sequence count'
result.index.name = 'Sample filename'
result.to_csv(..., header=True, index=True)
```